src/FramedClient.ts

```diff
@@ -223,14 +224,28 @@ class FramedClient {
                     delete this.requestCallbacks[data.requestId];
                 }
             } else if (event.data.eventName === PostMessageEventName.CreateObject) {
-                if (data.requestId in this.requestCallbacks) {
-                    this.requestCallbacks[data.requestId].resolve(data.response as AMSCreateObjectResponse);
-                    delete this.requestCallbacks[data.requestId];
+                if (data.eventStatus === PostMessageEventStatus.Success) {
+                    if (data.requestId in this.requestCallbacks) {
+                        this.requestCallbacks[data.requestId].resolve(data.response as AMSCreateObjectResponse);
+                        delete this.requestCallbacks[data.requestId];
+                    }
+                } else {
+                    if (data.requestId in this.requestCallbacks) {
+                        this.requestCallbacks[data.requestId].reject();
+                        delete this.requestCallbacks[data.requestId];
+                    }
                 }
             } else if (event.data.eventName === PostMessageEventName.UploadDocument) {
-                if (data.requestId in this.requestCallbacks) {
-                    this.requestCallbacks[data.requestId].resolve(data.response as FileMetadata);
-                    delete this.requestCallbacks[data.requestId];
+                if (data.eventStatus === PostMessageEventStatus.Success) {
+                    if (data.requestId in this.requestCallbacks) {
+                        this.requestCallbacks[data.requestId].resolve(data.response as FileMetadata);
+                        delete this.requestCallbacks[data.requestId];
+                    }
+                } else {
+                    if (data.requestId in this.requestCallbacks) {
+                        this.requestCallbacks[data.requestId].reject();
+                        delete this.requestCallbacks[data.requestId];
+                    }
                 }
             } else if (event.data.eventName === PostMessageEventName.GetViewStatus) {
                 if (data.requestId in this.requestCallbacks) {

```

Comment:
Are we not adding the success callback condition for ViewStautus event ? Do we use GetVewStatus anywhere in the attachment flow, like in middleware for checking the upload status

<img width="661" alt="image" src="https://github.com/microsoft/omnichannel-amsclient/assets/49963796/7b989ed7-aae3-4bc9-b303-ef200978f3b1">

Reply:
even calls like viewStatus failed, download flow should continue. But for upload flow, if at one point we got 400 but still try to send the message C1, that is data loss on C2 side, since it's marked as sent success. So this kind of error handling is not necessary for download for now, but needs to be there for upload